src/Control/Monad/Cofree/Trans.purs

```diff
@@ -0,0 +1,156 @@
+-- | This module defines a lazy implementation of the _cofree monad transformer_.
+
+-- | Given a `CofreeT` f m a:
+-- | - 'f' is a `Functor`, generally representing an AST,
+-- | - 'm' is a 'Monad', generally representing an effect,
+-- | - and 'a' is the type of the annotation.
+-- |
+-- | Usually, you would use `CofreeT` to annotate an existing AST with
+-- | metadata such as source locations, file names, etc.
+
+module Control.Monad.Cofree.Trans where
+
+import Prelude
+
+import Control.Alternative (class Alt, class Plus, empty, (<|>))
+import Control.Apply (lift2)
+import Control.Comonad (class Comonad, class Extend, extend, extract)
+import Control.Comonad.Cofree.Class (class ComonadCofree)
+import Control.Comonad.Env.Class (class ComonadAsk, ask)
+import Control.Comonad.Trans.Class (class ComonadTrans)
+import Control.Monad.Trans.Class (class MonadTrans)
+import Data.Bifunctor (bimap)
+import Data.Foldable (class Foldable, foldMap, foldlDefault, foldrDefault)
+import Data.Function (on)
+import Data.Traversable (class Traversable, sequenceDefault, traverse)
+import Data.Tuple (Tuple(..))
+import Data.Tuple as T
+import Effect.Aff.Class (class MonadAff, liftAff)
+import Effect.Class (class MonadEffect, liftEffect)
+
+-- | The cofree monad transformer for the functor 'f'.
```

Comment:
_monad_ transformer? Do you mean comonad transformer? I know it's also a monad transformer, but it's hardly the main use case.

Reply:
Also, in that light, it's customary to name the argument `w`, not `m`.

Reply:
Also because in a category theoretic sense, it is the cofree comonad transformer in the sense that it's an adjoint to the forgetful functor to natural transformations.